### DISCUSSION TITLE
Italic and Bold font style does not work on MacOS

### DISCUSSION THREAD

**Discussion by hanslavdal** [skia4delphi/skia4delphi#325] [General]
Thank you for the great work you have made with skia4delphi! Its great that we can now finally enable Metal on MacOS without quality degration.
However, there seem to be a general issue on MacOS that the Italic and Bold font style does not work. We mainly use font styles with the  "DelphiHtmlComponents" and reported the error there, but Alexander answered that it was a general problem in Skia, and after that I tried with a simple application and can comfirm that its the case.
I don't know if you are you aware of this bug already? 

PS. There is an report on quality portal about this too: https://embt.atlassian.net/servicedesk/customer/portal/1/RSS-1231

**Comment by viniciusfbb** [comment]
Hi @hanslavdal, thank you for the report!

We already started an issue for it, take a look: https://github.com/skia4delphi/skia4delphi/issues/322

**Comment by viniciusfbb** [comment]
@hanslavdal According @jareddavison in https://github.com/skia4delphi/skia4delphi/issues/322#issuecomment-2227230862 this issue is not reproducible.

Are you using a custom font? If so, you should also register the bold version of your custom font. If not, can you give more information about device, OS, RAD Studio version / skia version?

**Comment by jareddavison** [comment]
I've tested ThtPanel for this. From what I can see it shows bold if TLabel shows bold:

![image](https://github.com/user-attachments/assets/cee686b2-4361-47b8-a614-880c4384122d)

**Reply by jareddavison** [reply]
This is the same program except with the patch suggested in #322 
![image](https://github.com/user-attachments/assets/19c572f9-671d-43b9-b4cc-f4bcc313ed04)


@hanslavdal , one thought I have is maybe you could be missing the latest updates from HtmlComponents. Maybe consider re-downloading the latest hcl48release and update in case that is part of your issue. (I have found that the content at the download url had been updated, although the version number not changed)

**Comment by jareddavison** [comment]
@hanslavdal , I'm aware of some additional issues with ThtPanel which are currently unresolved, but I've made Alexander is aware and requested help with.

E.g Herculanum font. (examples use the patched FMX.SKIA.Canvas.pas)

You'll see SKIA has defaulted back to ordinary font for non regular style, whereas ThtPanel has just used the regular for all the others, but they aren't styled.

![image](https://github.com/user-attachments/assets/e1937d14-d99a-4d89-80ba-423248e0692d)

E.g Klee

![image](https://github.com/user-attachments/assets/a51c8f8f-2408-44dc-bb28-8ea3efb2a839)

Another custom font Happy Times: (weirdness there too, not just MACOS but windows also (without SKIA))
![image](https://github.com/user-attachments/assets/2e8a146a-8c2d-45d5-bfea-458c62c3492e)

**Comment by jareddavison** [comment]
Using Bodoni 72 which is available in Apple Pages, I've done a comparison.

There does seem to be a problem with SKIA choosing the correct font for non-regular styles (even with the patch), for the TLabel components. You will see ThtPanel seems to have done a better job somehow in this case, although its hard to tell in the Italic & Bold case if it is bold, it doesn't appear to be when compared to bold line above, but Apple Pages also failed to Italic the Bold and Italic lines at the bottom too.

![image](https://github.com/user-attachments/assets/c015bc62-94f5-408a-b473-8b9f814296bf)


Also without SKIA it behaves similarly although the italics styling seems different in the TLabel:

![image](https://github.com/user-attachments/assets/895555f9-2e5b-4aad-bc55-bd5221229824)

**Reply by viniciusfbb** [reply]
@jareddavison If you are using a custom font, you should register all .ttf of all styles that you will use. When you are downloading the font, you should download for example the .ttf of: Regular, Regular + Italic, Bold, Bold + Italic, etc, and register each one using as documented here: https://github.com/skia4delphi/skia4delphi/blob/main/Documents/LABEL.md#fmx

**Reply by jareddavison** [reply]
> @jareddavison If you are using a custom font, you should register all .ttf of all styles that you will use. When you are downloading the font, you should download for example the .ttf of: Regular, Regular + Italic, Bold, Bold + Italic, etc, and register each one using as documented here: https://github.com/skia4delphi/skia4delphi/blob/main/Documents/LABEL.md#fmx

@viniciusfbb all the fonts demonstrated are from the enumeration of the installed operating system fonts so these should be automatically loaded by fmx and skia is my understanding.

**Comment by hanslavdal** [comment]
@viniciusfbb I use the latest version of HTML components (48). But for my test, I just use a TLabel in a new empty app (with Skia enabled), using the default font:
![image](https://github.com/user-attachments/assets/f96e9461-5954-4963-95d9-10170a53b693)

I saw there is a workaround mentioned in the [other thread](https://github.com/skia4delphi/skia4delphi/issues/322), but since I am using the build-in Skia in Delphi 12.1, I am not sure how many dependencies there are if I rebuild it.

PS. Bold font DOES work, sorry that I included that one in my message. I misread my own notes: the problem we had with bold font was that it appeared one pixel higher than the normal text, but I cannot reproduce that right now. So its only Italic that fails.

**Reply by viniciusfbb** [reply]
@hanslavdal No worry, we should ensure that everything is working fine and all reports are important.

> the problem we had with bold font was that it appeared one pixel higher than the normal text

This can be related to the font weight. On non-windows platforms, the FMX does not support font weight, so it only works with Regular (400) or Bold (700). If you have a font weight below Bold (700), for example Semibold (600) or Medium (500), the FMX without Skia will render as Regular (400), while FMX + Skia will render as the font weight chosen, in this case a thickness between Regular and Bold. This can explain your trouble.

**Reply by jareddavison** [reply]
> I saw there is a workaround mentioned in the [other thread](https://github.com/skia4delphi/skia4delphi/issues/322), but since I am using the build-in Skia in Delphi 12.1, I am not sure how many dependencies there are if I rebuild it.

If you aren't using runtime packages in your project it easy to try.  Open up the version in Delphi 12.1 of 
FMX.SKIA.Canvas.pas and save it into you project locally and patch it. That's all.

**Reply by hanslavdal** [reply]
I can confirm that it worked just by patching this one file and including it with our main application. Thank you very much for your quick assistance!
